File: Python/0551.py
class Solution:
    def checkRecord(self, s: str) -> bool:
        absence = 0
        i = 0
        n = len(s)

        while i < n:
            if s[i] == 'A':
                absence += 1
                if absence == 2:
                    return False
                i += 1
            elif s[i] == 'L':
                consecutiveLate = 1
                i += 1
                while i < n and s[i] == 'L':
                    consecutiveLate += 1
                    i += 1
                if consecutiveLate >= 3:
                    return False
            else:
                i += 1

        return True

File: C/0551.c
#include <string.h>

bool checkRecord(char* s) {
    int i = 0, absence = 0, n = strlen(s);

    while (i < n) {
        if (s[i] == 'A') {
            absence++;
            if (absence == 2) {
                return 0;
            }
            i++;
        } else if (s[i] == 'L') {
            int consecutiveLate = 1;
            i++;
            while (i < n && s[i] == 'L') {
                consecutiveLate++;
                i++;
            }
            if (consecutiveLate >= 3) {
                return 0;
            }
        } else {
            i++;
        }
    }

    return 1;
}
